packages/python/m/github/argumets.py

```diff
@@ -0,0 +1,34 @@
+import re
+
+from m.git import get_remote_url
+from pydantic import BaseModel, Field
+
+
+def repo_and_owner():
+    repository_url = get_remote_url()
+
+    if repository_url.is_bad:
+        return None, None
+
+    repository_url_parser = re.compile(r"((?:https?:\/\/)|(?:\w+@))github\.com[:\/]([^\/]+)\/([^\/]+)\.git")
+
+    match = repository_url_parser.match(repository_url.value)
+
+    if match is None:
+        return None, None
+
+    _, owner, repo = match.groups()
+
+    return owner, repo
+
+owner, repo = repo_and_owner()
+
+class RepoAndOwnerArgs(BaseModel):
+    owner: str = Field(
```

Comment:
Instead of evaluating the `repo_and_owner` function we can make it as a validator, similar to this:

https://github.com/jmlopez-rod/m/blob/master/packages/python/m/cli/validators.py#L9

In this way you make one validator for each and raise an argparse exception when the argument is not provided and we are executing the cli from a non-git repo.

To use the validator you can do something similar to

https://github.com/jmlopez-rod/m/blob/master/packages/python/m/cli/commands/ci/celt.py#L70

We can default to `@-` to read from the git config, otherwise if its not set to the default then we use the use argument.

That way we don't execute the repo_and_owner whenever we load the `argumets` (typo in file name, check that one out).